lib/dotcom/alerts/endpoint_stops.ex

```diff
@@ -0,0 +1,155 @@
+defmodule Dotcom.Alerts.EndpointStops do
+  @moduledoc """
+  Module for determining endpoint-stop ranges affected by an alert
+  """
+
+  alias Alerts.Alert
+  alias Dotcom.Alerts.EndpointStops.Behaviour
+
+  @route_patterns_repo Application.compile_env!(:dotcom, :repo_modules)[:route_patterns]
+  @routes_repo Application.compile_env!(:dotcom, :repo_modules)[:routes]
+  @stops_repo Application.compile_env!(:dotcom, :repo_modules)[:stops]
+
+  @behaviour Behaviour
+
+  @impl Behaviour
+  def endpoint_stops(alerts, route_ids) do
+    alerts
+    |> Enum.map(&endpoint_stops_for_alert(&1, route_ids))
+    |> Enum.reject(&Kernel.is_nil/1)
+  end
+
+  defp endpoint_stops_for_alert(alert, route_ids) do
+    affected_stop_ids = alert |> Alert.get_entity(:stop)
+    affected_direction_id = alert |> affected_direction_id()
+
+    direction_names =
+      route_ids
+      |> as_actual_route_ids()
+      |> List.first()
+      |> @routes_repo.get()
+      |> Kernel.then(& &1.direction_names)
+      |> to_forward_backward_direction_map(affected_direction_id)
+
+    if MapSet.size(affected_stop_ids) < 2 do
+      nil
+    else
+      route_ids
+      |> affected_stop_lists_for(affected_direction_id, affected_stop_ids)
+      |> drop_sublists()
+      |> to_endpoints(direction_names)
+      |> case do
+        endpoints -> endpoints
+      end
```

Comment:
Should this do something? Does this do something? :O

Reply:
Uhhh, hum... what was I thinking here? I'm guessing that I started out with some more interesting behavior where this was sometimes not just `endpoints`, but then incompletely cleaned it up.

Now it is [more completely cleaned up](https://github.com/mbta/dotcom/pull/2558/commits/8097b778ae98a5de6a454eb5a30093d04b32e9fb)